src/script/ceph-backport.sh

```diff
@@ -111,115 +122,233 @@
 # Nathan
 #
 source $HOME/bin/backport_common.sh
+this_script=$(basename "$0")
+verbose=
+
+if [[ $* == *--debug* ]]; then
+    set -x
+    verbose="1"
+fi
+
+if [[ $* == *--verbose* ]]; then
+    verbose="1"
+fi
 
-function failed_required_variable_check () {
-    local varname=$1
-    echo "$0: $varname not defined. Did you create $HOME/bin/backport_common.sh?"
-    echo "(For instructions, see comment block at beginning of script)"
+function log {
+    local msg="$1"
+    local level="$2"
```

Comment:
nit (can be ignored if you don't like): why not use level as the first param? This is rather conventional (`log(level, mesage args ...)`). 
Then also you could do:
```
local level="$1"
shift
msg="$@"
```
And then when using it would not be necessary to quote the message, and something like this would work:
```
log verbose Checking mandatory variables
```
Actually I see the reason why you use level as the second param: because it is optional. I would then just add a wrapper function, something like `out() { log '' $@ }` and use it for such cases (the same way you could add `warn`, `err`, `debug` wrappers).

Reply:
very nice suggestion, @trociny - implemented it